src/GitHub.TeamFoundation.14/Services/VSGitExt.cs

```diff
@@ -95,19 +122,23 @@ bool TryInitialize()
             return false;
         }
 
+        async Task<T> GetServiceAsync<T>() where T : class
+        {
+            // GetService must be called from the Main thread.
+            await ThreadingHelper.SwitchToMainThreadAsync();
+            return serviceProvider.GetService<T>();
```

Comment:
The way we're exposing MEF implementations as services is by having a MEF shim that redirects to the real instance instead, like we do for [GitHubServiceProvider](https://github.com/github/VisualStudio/blob/docs/clarify-tokens/src/GitHub.VisualStudio/Services/GitHubServiceProvider.cs#L32), UsageTracker (https://github.com/github/VisualStudio/blob/docs/clarify-tokens/src/GitHub.VisualStudio/Services/UsageTrackerDispatcher.cs and https://github.com/github/VisualStudio/blob/docs/clarify-tokens/src/GitHub.VisualStudio/Services/UsageTracker.cs) and registering the service in the `ServiceProviderPackage`. This is how this one should also probably be registered so we don't have to keep switching threads to get at it.

Reply:
I'll do something similar in this PR. I can avoid switching threads by using `IAsyncServiceProvider` to retrieve `IGitExt`.

Reply:
I've done as you've suggested in #1510. This PR is now obsolete.